packages/@glimmer/integration-tests/test/precompile-test.ts

```diff
@@ -32,75 +24,56 @@ QUnit.module('templateFactory', {
 });
 
 QUnit.test('generates id if no id is on the serialized template', (assert) => {
-  let factory1 = templateFactory(serializedTemplateNoId);
-  let factory2 = templateFactory(serializedTemplateNoId);
-  assert.ok(factory1.id, 'is present');
-  assert.ok(factory2.id, 'is present');
+  let factory1 = templateFactory(serializedTemplateNoId) as TemplateFactoryWithIdAndMeta;
+  let factory2 = templateFactory(serializedTemplateNoId) as TemplateFactoryWithIdAndMeta;
+  assert.ok(factory1.__id, 'is present');
+  assert.ok(factory2.__id, 'is present');
   assert.notEqual(
-    factory1.id,
-    factory2.id,
+    factory1.__id,
+    factory2.__id,
     'factories without underlying id create new id per factory'
   );
 });
 
 QUnit.test('id of template matches factory', (assert) => {
-  let factory = templateFactory(serializedTemplate);
-  let template = unwrapTemplate(factory.create());
+  let factory = templateFactory(serializedTemplate) as TemplateFactoryWithIdAndMeta;
+  let template = unwrapTemplate(factory()) as TemplateWithIdAndReferrer;
   assert.ok(template.id, 'is present');
-  assert.equal(template.id, factory.id, 'template id matches factory id');
+  assert.equal(template.id, factory.__id, 'template id matches factory id');
 });
 
 QUnit.test('meta is accessible from factory', (assert) => {
-  let factory = templateFactory(serializedTemplate);
-  assert.deepEqual(factory.meta, {
-    kind: 'unknown',
-    meta: {},
-    module: 'template/module/name',
-    name: 'default',
-  });
+  let factory = templateFactory(serializedTemplate) as TemplateFactoryWithIdAndMeta;
+  assert.deepEqual(factory.__meta, { moduleName: 'template/module/name' });
 });
 
 QUnit.test('meta is accessible from template', (assert) => {
   let factory = templateFactory(serializedTemplate);
-  let template = unwrapTemplate(factory.create());
+  let template = unwrapTemplate(factory()) as TemplateWithIdAndReferrer;
   assert.deepEqual(
-    template.referrer as AnnotatedModuleLocator,
-    {
-      kind: 'unknown',
-      meta: {},
-      module: 'template/module/name',
-      name: 'default',
-    },
+    template.referrer,
+    { moduleName: 'template/module/name', owner: null },
     'template has expected meta'
   );
 });
 
-QUnit.test('can inject per environment things into meta', (assert) => {
+QUnit.test('can inject owner into template', (assert) => {
   let owner = {};
-  let factory = templateFactory<AnnotatedModuleLocator>(serializedTemplate);
+  let factory = templateFactory(serializedTemplate);
 
-  let template = unwrapTemplate(
-    factory.create(
-      assign({}, DEFAULT_TEST_META, {
-        owner,
-      })
-    )
-  );
+  let template = unwrapTemplate(factory(owner)) as TemplateWithIdAndReferrer;
 
   assert.strictEqual(template.referrer.owner, owner, 'is owner');
   assert.deepEqual(
-    template.referrer as AnnotatedModuleLocator & { owner: unknown },
+    template.referrer,
```

Comment:
is this used? you made referrer owner in the lookup API

Reply:
It's unclear if this API is still used, but I've seen references to it in a few places in code that use `__meta`. I decided to keep it around as a backwards compatible API for the time being, and remove it in the near future. I added a TODO comment to track removing it.